src/Domain/Item/ItemService.ts

```diff
@@ -0,0 +1,75 @@
+import * as dayjs from 'dayjs'
+import { inject, injectable } from 'inversify'
+import TYPES from '../../Bootstrap/Types'
+import { GetItemsDTO } from './GetItemsDTO'
+
+import { GetItemsResult } from './GetItemsResult'
+import { Item } from './Item'
+import { ItemQuery } from './ItemQuery'
+import { ItemRepositoryInterface } from './ItemRepositoryInterface'
+import { ItemServiceInterface } from './ItemServiceInterface'
+import { SaveItemsResult } from './SaveItemsResult'
+
+@injectable()
+export class ItemService implements ItemServiceInterface {
+  private readonly DEFAULT_ITEMS_LIMIT = 100000
+  private readonly SYNC_TOKEN_VERSION = 2
+
+  constructor (
+    @inject(TYPES.ItemRepository) private itemRepository: ItemRepositoryInterface,
+  ) {
+  }
+
+  async getItems(dto: GetItemsDTO): Promise<GetItemsResult> {
+    const lastSyncTime = this.getLastSyncTime(dto)
+
+    const itemQuery: ItemQuery = {
+      userUuid: dto.userUuid,
+      lastSyncTime,
+      contentType: dto.contentType,
+      deleted: lastSyncTime ? undefined : false,
+      sortBy: 'updatedAt',
+      limit: !dto.limit || dto.limit < 1 ? this.DEFAULT_ITEMS_LIMIT : dto.limit
+    }
+
+    const items = await this.itemRepository.findAll(itemQuery)
+
+    const lastSyncTimeWithPrecautiousMicrosecond = (items[items.length - 1].updatedAt + 1) / 1000000
```

Comment:
Is the closing parenthesis here misplaced or maybe I'm reading it wrong? Should it be:

```
items[items.length - 1].updatedAt + (1 / 1000000)
```

or is that the same thing?

Reply:
Oh nevermind. updatedAt is already an integer timestamp so adding 1 is adding 1 microsecond.